src/custom/utils/operator.ts

```diff
@@ -206,16 +224,16 @@ function toApiAddress(address: string, chainId: ChainId): string {
 async function _getJson(chainId: ChainId, url: string): Promise<any> {
   let response: Response | undefined
   try {
-    const responseMaybeOk = await _fetchGet(chainId, url)
-    response = responseMaybeOk.ok ? responseMaybeOk : undefined
-  } catch (error) {
-    // do nothing
-  }
-
-  if (!response) {
-    throw new Error('Error getting the fee')
-  } else {
-    return response.json()
+    response = await _fetchGet(chainId, url)
+  } finally {
+    if (!response) {
+      throw new Error(`Error getting query @ ${url}`)
+    } else if (!response.ok) {
+      const errorResponse = await response.json()
+      throw errorResponse
```

Comment:
I think in this case you are not using `_getErrorForUnsuccessfulPostOrder`, shouldn't you?


I maybe would create a custom error https://stackoverflow.com/questions/41102060/typescript-extending-error-class

This way you can throw your custom error, with a nice message, and also the original data of the response so it can also be handled

Reply:
done!